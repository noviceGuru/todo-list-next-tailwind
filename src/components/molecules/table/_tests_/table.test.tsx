import { render, screen, within, cleanup } from "@testing-library/react/pure"
import '@testing-library/jest-dom'
import Table from "../table"

render(<Table
	initialData={[
		{
			"id": 1,
			"task": "write tests"
		},
		{
			"id": 2,
			"task": "make the storybook"
		}
	]}
/>)

it('the table renders row correctly with the data and correct looks', () => {
	const table = screen.getByRole('table')

	expect(table).toBeInTheDocument()
	expect(table).toMatchSnapshot()
})

it('renders initial buttons (delete and edit) correctly', () => {

	const table = screen.getByRole('table')
	const row1 = within(table).getByRole('row', {
		name: 'write tests Delete Edit'
	})

	const button_delete1 = within(row1).getByRole('button', {
		name: /delete/i
	})

	const button_edit1 = within(row1).getByRole('button', {
		name: /edit/i
	})

	expect(button_delete1).toBeInTheDocument()
	expect(button_edit1).toBeInTheDocument()

	
	const row2 = within(table).getByRole('row', {
		name: 'write tests Delete Edit'
	})

	const button_delete2 = within(row2).getByRole('button', {
		name: /delete/i
	})

	const button_edit2 = within(row2).getByRole('button', {
		name: /edit/i
	})

	expect(button_delete2).toBeInTheDocument()
	expect(button_edit2).toBeInTheDocument()
	cleanup()
})
