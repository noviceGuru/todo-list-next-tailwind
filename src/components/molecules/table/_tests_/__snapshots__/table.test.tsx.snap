// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the table renders row correctly with the data and correct looks 1`] = `
<table
  class="border-2 w-96"
>
  <thead>
    <tr
      class="border-2 border-black rounded"
    >
      <th>
        a todo table
      </th>
      <td
        class="border-l-2 border-black flex justify-center"
      >
        <button
          class="border-2 rounded-md m-1 border-indigo-300 bg-indigo-300"
          type="button"
        >
          Add new
        </button>
      </td>
    </tr>
  </thead>
  <tbody>
    <tr
      class="border-2 border-indigo-400 "
    >
      <td
        class="border-4 border-pink-900 pl-2"
      >
        <span>
          write tests
        </span>
      </td>
      <td
        class="border-4 border-pink-800 bg-slate-300 w-32"
      >
        <button
          class="border-2 rounded-md m-1 bg-red-600 border-red-300"
          type="button"
        >
          Delete
        </button>
        <button
          class="border-2 rounded-md m-1 border-indigo-300 bg-indigo-300"
          type="button"
        >
          Edit
        </button>
      </td>
    </tr>
    <tr
      class="border-2 border-indigo-400 "
    >
      <td
        class="border-4 border-pink-900 pl-2"
      >
        <span>
          make the storybook
        </span>
      </td>
      <td
        class="border-4 border-pink-800 bg-slate-300 w-32"
      >
        <button
          class="border-2 rounded-md m-1 bg-red-600 border-red-300"
          type="button"
        >
          Delete
        </button>
        <button
          class="border-2 rounded-md m-1 border-indigo-300 bg-indigo-300"
          type="button"
        >
          Edit
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
